crates/ruff/src/rules/pylint/rules/empty_comment.rs

```diff
@@ -0,0 +1,86 @@
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::source_code::{Indexer, Locator};
+
+use crate::registry::Rule;
+use crate::settings::Settings;
+
+/// ## What it does
+/// Checks for useless empty comments.
+///
+/// ## Why is this bad?
+/// It is not an actual comment.
+///
+/// ## Example
+/// ```python
+/// print(1)  #
+/// ```
+///
+/// Use instead:
+/// ```python
+/// print(1)
+/// ```
+#[violation]
+pub struct EmptyComment;
+
+impl AlwaysAutofixableViolation for EmptyComment {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Line with empty comment")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Remove empty comment".to_string()
+    }
+}
+
+fn comment_part_of_string(line: &str, idx: usize) -> bool {
+    (line.get(..idx).unwrap().matches('"').count() % 2 == 1
+        && line.get(idx + 1..).unwrap().matches('"').count() % 2 == 1)
+        || (line.get(..idx).unwrap().matches('\'').count() % 2 == 1
+            && line.get(idx + 1..).unwrap().matches('\'').count() % 2 == 1)
+}
+
+fn is_line_commented(line: &str) -> bool {
+    if let Some(idx) = line.find('#') {
+        if comment_part_of_string(line, idx) {
+            is_line_commented(&format!(
+                "{}{}",
+                line.get(..idx).unwrap(),
+                line.get(idx + 1..).unwrap()
+            ))
+        } else {
+            true
+        }
+    } else {
+        false
+    }
+}
```

Comment:
I'm not entirely sure if I understand this code correctly but I think this can go wrong for multiline strings

```python
a = """A multiline string
that goes contains a # that looks like a comment""" #
```

There's no good way for you to detect this by just looking at the source (the same can happen for single quoted strings). 

From what I understand you're trying to achieve is to see if the comment is all empty. You can do this by using `trim_end_matches(|c| is_python_whitespace(c) || c == '#')`. This should trim all whitespace and `#` from the comment text. The comment is empty if the resulting string is `""`, and the comment is non empty if there's some text remaining.

Reply:
Let's hope my implementation is correct :sweat_smile: 

Our goal is to implement rules with the same semantic but we're free to diverge on how we implement the rule to get better performance or higher correctness.

Reply:
So, I'm trying out your suggestion but this logic will also trim `########` as an empty comment and `a = x # fooo #` also as an empty comment.

Reply:
The fist example makes sense to me. Should it only consider comments as empty if the # are separated by a space?


The second example is surprising to me. Can you share your code?

Reply:
The second example is in the test, oh woops:

```
print(b)  #
print(b)  # this is fine#
b = "#this#is#fine#"  # 
```

My assumption is that your you compare the trimmed line with the normal line, if anything was removed it has an empty comment? But maybe that's wrong :)

Reply:
I think that's too aggressive because it also is true if only part of a comment was removed. Can you try testing if the trimmed comment is empty (it won't sole the first problem)

Reply:
So, that would mean only useless comments on an a line would be stripped? That then only removes empty comments without any Python code?

Reply:
What I understand is that the rule should flag the following comments:

```python
# 
end_of_line #
end_of_line_nested # # # 
```

But it should not flag

```python
#########
"""abcd
efg # hij" 
efg # hij #" 
```

How about

```python
end_of_line ######
```

Assuming this is correct. The way I would approach the rule is by only looking at the comment text and ignoring the line entirely to detect empty comments (fixing requires detecting if it is an own-line or end of line comment)

```rust
let comment_text = locator.slice(comment_range);

if comment_text.trim_start_matches(|c| c == "#" || is_python_whitespace(c)).is_empty() {
	// create diagnostic
}
```

The condition is true if the comment only consists of whitespace or `#` (there are no remaining other characters). This incorrectly flags `######`. 

```rust
let comment_text = locator.slice(comment_range);
let mut rest = comment_text;

loop {
	rest = comment_text.trim_start_matches(is_python_whitespace).trim_start_matches('#');	

	if rest.is_empty() {
		// Comment consist only of `#...whitespace` sequences. 
		break;
	}

	// `##` or `# a`
	if !rest.starts_with(is_python_whitespace) {
		return;
	}
}

// create diagnostic
```